cpp/include/cuspatial/traj_thrust.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <ostream>
+#include <cuspatial/cuspatial.h>
+
+namespace cuspatial
+{
+
+	/**
+	 *@brief Thrust functor for comparing two TimeStamp variables; used in sorting based on timestamp
+	 */
+	__host__ __device__
+	inline bool operator<(const TimeStamp & t1,const TimeStamp & t2)
+	{
+		//cout<<"in operator<"<<endl;
+		if(t1.y<t2.y) return true;
+		else if((t1.y==t2.y)&&(t1.m<t2.m)) return true;
+		else if((t1.y==t2.y)&&(t1.m==t2.m)&&(t1.d<t2.d)) return true;
+		else if((t1.y==t2.y)&&(t1.m==t2.m)&&(t1.d==t2.d)&&(t1.hh<t2.hh)) return true;
+		else if((t1.y==t2.y)&&(t1.m==t2.m)&&(t1.d==t2.d)&&(t1.hh==t2.hh)&&(t1.mm<t2.mm)) return true;
+		else if((t1.y==t2.y)&&(t1.m==t2.m)&&(t1.d==t2.d)&&(t1.hh==t2.hh)&&(t1.mm==t2.mm)&&(t1.ss<t2.ss)) return true;
+		else if((t1.y==t2.y)&&(t1.m==t2.m)&&(t1.d==t2.d)&&(t1.hh==t2.hh)&&(t1.mm==t2.mm)&&(t1.ss==t2.ss)&&(t1.ms<t2.ms)) return true;
+		return false;
+	}
+
+	typedef  thrust::pair<TimeStamp, TimeStamp> TBBox;
+
+	/**
+	 *@brief Thrust functor for lifting TimeStamp to an interval (1D box)
+	 */
+	struct TBBox_transformation : public thrust::unary_function<TBBox,TBBox>
+	{
+		__host__ __device__
```

Comment:
Do you *ever* need to call this from the host? If not, make it device-only.

```suggestion
		__device__
```